logprep/connector/confluent_kafka/input.py

```diff
@@ -421,6 +434,26 @@ def _get_event(self, timeout: float) -> Union[Tuple[None, None], Tuple[dict, dic
             ) from error
         return event_dict, raw_event
 
+    def _add_input_connector_metadata_to_event(self, event_dict) -> Tuple[dict, Optional[str]]:
+        metadata = event_dict.get("_metadata", {})
+        for meta_field in ("last_partition", "last_offset"):
+            try:
+                del event_dict["_metadata"][meta_field]
+            except (TypeError, KeyError):
+                pass
```

Comment:
I have changed it, but I kept try except to check in case _metadata is not a dict, since this field might already exist in the event and be of any type. I did not check for dict directly, since it is more likely to not happen and more performant this way.